File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/events/PlainEventRenderer.java
package net.nemerosa.ontrack.model.events;

import net.nemerosa.ontrack.model.structure.*;

public class PlainEventRenderer implements EventRenderer {

    public static final EventRenderer INSTANCE = new PlainEventRenderer();

    @Override
    public String render(ProjectEntity projectEntity, Event event) {
        switch (projectEntity.getProjectEntityType()) {
            case PROJECT:
                return ((Project) projectEntity).getName();
            case BRANCH:
                return ((Branch) projectEntity).getName();
            case PROMOTION_LEVEL:
                return ((PromotionLevel) projectEntity).getName();
            case VALIDATION_STAMP:
                return ((ValidationStamp) projectEntity).getName();
            case BUILD:
                return ((Build) projectEntity).getName();
            case VALIDATION_RUN:
                return "#" + ((ValidationRun) projectEntity).getRunOrder();
            default:
                throw new EventCannotRenderEntityException(event.getTemplate(), projectEntity);
        }
    }

    @Override
    public String render(String valueKey, String value, Event event) {
        return value;
    }
}

File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/events/PlainEventRenderer.kt
package net.nemerosa.ontrack.model.events

import net.nemerosa.ontrack.model.structure.ProjectEntity
import net.nemerosa.ontrack.model.structure.displayName
import org.springframework.stereotype.Component

@Component
class PlainEventRenderer : AbstractEventRenderer() {

    override val id: String = "text"
    override val name: String = "Text"

    override fun render(projectEntity: ProjectEntity): String = projectEntity.displayName

    override fun renderStrong(value: String): String = value

    override fun renderLink(text: String, href: String): String = text

    override fun renderList(list: List<String>): String =
        list.joinToString("\n") { "* $it" }

    override fun renderSpace(): String = "\n\n"

    override fun renderSection(title: String, content: String): String =
        """
            |$title
            |
            |$content
        """.trimMargin()

    companion object {
        @JvmField
        val INSTANCE: EventRenderer = PlainEventRenderer()
    }
}